setup.sh

```diff
@@ -1,25 +1,35 @@
 #!/bin/bash
 
-echo "Instaling Bundler"
-gem install bundler
+# Tells the shell script to exit if it encounters an error
+set -e
+
+# -- Log -----------------------------------------------------------------------
+function msg { echo  "\033[0;37m$1\033[0m"; }
+function msg_ok { echo  "➜\033[1;32m $1 ✔\033[0m"; }
+function msg_run { echo  "➜\033[1;35m $1 $ $2\033[0m"; }
+
+# -- Dependencies --------------------------------------------------------------
+echo "Instaling Bundler!"
+sudo gem install bundler
 bundle install
 
-echo "Instaling or Updating Homebrew"
-which -s brew
-if [[ $? != 0 ]] ; then
-    # Install Homebrew
-    ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+echo "Installing Homebrew!"
+if which brew &> /dev/null; then
+	msg_ok "homebrew"
 else
-    brew update
+	msg_run "homebrew" "/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)""
+	/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
```

Comment:
Aqui o uso de `/usr/bin/ruby` está forçando o uso do ruby da máquina. Será que talvez também não seja melhor deixar "solto"? Para casos de ambientes virtuais?